py_ecc/typing.py

```diff
@@ -64,8 +65,8 @@
     bls12_381_FQ2,
     bls12_381_FQ12,
 )
-Point2D = Tuple[Field, Field]
-Point3D = Tuple[Field, Field, Field]
+Point2D = Optional[Tuple[Field, Field]]  # Point at infinity is encoded as a None
```

Comment:
This is the meat of this PR and addresses 48 type errors.

Side note, should type checking be a part of CI on this repo? I feel like it should.

Reply:
I opened issue #95 for this.